test/conformance/api/v1/util.go

```diff
@@ -49,12 +55,25 @@ func checkForExpectedResponses(ctx context.Context, t testing.TB, clients *test.
 }
 
 func validateDomains(t testing.TB, clients *test.Clients, baseDomain *url.URL,
-	baseExpected, trafficTargets, targetsExpected []string) error {
-	subdomains := make([]*url.URL, len(trafficTargets))
-	for i, target := range trafficTargets {
-		subdomain, _ := url.Parse(baseDomain.String())
-		subdomain.Host = target + "-" + baseDomain.Host
-		subdomains[i] = subdomain
+	baseExpected []string, trafficExpectedPairs []trafficExpectedPair, serviceName string) error {
+
+	service, err := clients.ServingClient.Services.Get(context.Background(), serviceName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not query service traffic status: %w", err)
+	}
+	subdomains := make([]*url.URL, len(trafficExpectedPairs))
+	for i, pair := range trafficExpectedPairs {
+		found := false
+		for _, traffic := range service.Status.Traffic {
```

Comment:
nit: as a readability optimization, this could be done by creating a map of the service.Status.Traffic array with tag:url and then iterating over trafficExpectedPairs